core/src/main/java/org/infinispan/protostream/annotations/impl/ProtoMessageTypeMetadata.java

```diff
@@ -339,7 +339,7 @@ private void discoverFields(XClass clazz, Set<XClass> examinedClasses) {
                   // MESSAGE is the default and stands for 'undefined', we can override it with a better default
                   protobufType = Type.BYTES;
                }
-               if (!javaType.isArray() && !javaType.isPrimitive() && javaType.isAbstract()) {
+               if (!javaType.isArray() && !javaType.isPrimitive() && (javaType.isAbstract() && !javaType.isEnum())) {
```

Comment:
OMG, unneeded grouping with ( ) !!

I think this fix is needed on line 493 also, but no existing test highlights that case.

Reply:
I've updated `Simple.java` so that the method variation is also covered by the testsuite.